### PR TITLE
Faucet mechanism disabled option

### PR DESCRIPTION
**Pull request by bogdan-rosianu**
Added the option to disable the faucet mechanism by simply setting the value to 0 from config. Also applied this as the default action.

### CODE REVIEW

File: api/mock/facade.go [hunk 1]
```diff
@@ -28,6 +29,15 @@ type Facade struct {
 	GetTransactionByHashAndSenderAddressHandler func(txHash string, sndAddr string) (*transaction.ApiTransactionResult, int, error)
 }
 
+// IsFaucetEnabled -
+func (f *Facade) IsFaucetEnabled() bool {
+	if f.IsFaucetEnabledHandler != nil {
+		return f.IsFaucetEnabledHandler()
+	}
+
+	return true
```
Comment (miiu96) [inline api/mock/facade.go:38]: default false maybe ?

File: api/transaction/routes_test.go [hunk 2]
```diff
@@ -414,3 +414,29 @@ func TestSendUserFunds_CorrectValue(t *testing.T) {
 
 	assert.Equal(t, expectedValue, callValue)
 }
+
+func TestSendUserFunds_FaucetNotEnabled(t *testing.T) {
+	t.Parallel()
+	receiver := "05702a5fd947a9ddb861ce7ffebfea86c2ca8906df3065ae295f283477ae4e43"
```
Comment (miiu96) [inline api/transaction/routes_test.go:420]: add empty line before

File: facade/elrondProxyFacade.go [hunk 3]
```diff
@@ -121,6 +121,10 @@ type networkConfig struct {
 	minTransactionVersion uint32
 }
 
+func (epf *ElrondProxyFacade) IsFaucetEnabled() bool {
```
Comment (miiu96) [inline facade/elrondProxyFacade.go:124]: add comment

File: facade/mock/faucetProcessorStub.go [hunk 4]
```diff
@@ -8,11 +8,20 @@ import (
...
 		receiver string, value *big.Int, chainID string, version uint32) (*data.Transaction, error)
 	SenderDetailsFromPemCalled func(receiver string) (crypto.PrivateKey, string, error)
 }
 
+func (fps *FaucetProcessorStub) IsEnabled() bool {
+	if fps.IsEnabledCalled != nil {
+		return fps.IsEnabledCalled()
+	}
+
+	return true
```
Comment (miiu96) [inline facade/mock/faucetProcessorStub.go:22]: default false ?

File: process/factory/faucetProcessorFactory.go [hunk 5]
```diff
@@ -0,0 +1,25 @@
+package factory
```
Comment (miiu96) [inline process/factory/faucetProcessorFactory.go:1]: nice implementation

File: process/faucetProcessor.go [hunk 6]
```diff
@@ -85,6 +85,11 @@ func NewFaucetProcessor(
 	}, nil
 }
 
+// IsEnabled returns true if
+func (fp *FaucetProcessor) IsEnabled() bool {
```
Comment (miiu96) [inline process/faucetProcessor.go:89]: finish comment

File: testing/testHttpServer.go [hunk 7]
```diff
@@ -172,8 +173,9 @@ func (ths *TestHttpServer) processRequestGetConfigMetrics(rw http.ResponseWriter
 		"erd_round_duration":             30,
 		"erd_shard_consensus_group_size": 30,
 		"erd_start_time":                 30,
+		"erd_min_transaction_version":    1,
 	}
-	resp := data.GenericAPIResponse{Data: responseStatus, Code: data.ReturnCodeSuccess}
+	resp := data.GenericAPIResponse{Data: gin.H{"config": responseStatus}, Code: data.ReturnCodeSuccess}
```
Comment (miiu96) [inline testing/testHttpServer.go:178]: why you use gin.H here ?

File: api/mock/facade.go (hunk 1 above)
Comment (bogdan-rosianu) [reply api/mock/facade.go:38]: it can remain true as to simulate the situation before this PR

File: api/transaction/routes_test.go (hunk 2 above)
Comment (bogdan-rosianu) [reply api/transaction/routes_test.go:420]: done

File: facade/elrondProxyFacade.go (hunk 3 above)
Comment (bogdan-rosianu) [reply facade/elrondProxyFacade.go:124]: done

File: facade/mock/faucetProcessorStub.go (hunk 4 above)
Comment (bogdan-rosianu) [reply facade/mock/faucetProcessorStub.go:22]: no. same reason as above

File: process/faucetProcessor.go (hunk 6 above)
Comment (bogdan-rosianu) [reply process/faucetProcessor.go:89]: done

File: testing/testHttpServer.go (hunk 7 above)
Comment (bogdan-rosianu) [reply testing/testHttpServer.go:178]: as to map the `data` field to `key: value`. the way it was before didn't work and reading network configuration failed.

**Pull request merged**